### PR TITLE
reckon: csv to ledger command line application

### PR DESCRIPTION
**Pull request by aljce**
###### Motivation for this change

Adds a ruby script that can transform CSV files into ledger files.
###### I need help

I dont program in ruby and so I just followed the nixpkg manual on how to build ruby applications.
When I run `nix-build -A reckon` I get the following error:

``` sh
error: attribute ‘source’ missing, at /home/kyle/packages/pkgs/development/ruby-modules/bundler-env/default.nix:35:53
```

Can anyone help me complete this pr?
###### Things done
- [x] Tested using sandboxing
  ([nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS,
    or option `build-use-chroot` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by FRidh** [comment]
@zimbatm is the ruby expert? :)

File: pkgs/tools/text/reckon/gemset.nix [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  chronic = {
+  };
+  fastercsv = {
+  };
+  highline = {
+  };
+  reckon = {
+  };
+  terminal-table = {
+  };
```
Comment (zimbatm) [inline pkgs/tools/text/reckon/gemset.nix]: there should be versions and hashes in there. How did you generate it?

File: pkgs/tools/text/reckon/default.nix [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  gemfile = ./Gemfile;
+  lockfile = ./Gemfile.lock;
+  gemset = ./gemset.nix;
+ 
+  meta = with lib; {
+    description = "Flexibly import bank account CSV files into Ledger for command line accounting";
+    license = licenses.mit;
+    maintainers = "mckean.kylej@gmail.com";
+    platforms = platforms.unix;
+  };
```
Comment (zimbatm) [inline pkgs/tools/text/reckon/default.nix]: There is an issue right now with bundlerEnv where other binaries are being exported as well. Check out the `cide` package to see how to only keep what you want.

File: pkgs/tools/text/reckon/gemset.nix (hunk 1 above)
Comment (mckeankylej) [inline pkgs/tools/text/reckon/gemset.nix]: Turns out when you run bundix in a nix-shell it needs the ssl cert export patch applied to the shell or bundix cant download things. This is fixed in latest commit.

File: pkgs/tools/text/reckon/default.nix (hunk 2 above)
Comment (mckeankylej) [inline pkgs/tools/text/reckon/default.nix]: Fixed in latest commit.

**Comment by mckeankylej** [comment]
I fixed all the errors, can someone review my pr? @zimbatm 
I am pretty sure this is good to merge.

**Comment by zimbatm** [comment]
@mckeankylej sorry for not being responsive on that one.
Just squashed all the commits and fixed the commit title to `reckon: init at 0.4.4` and pushed to master as de7a2651173655eef7c37369a05b6d4a47b64f41

**Comment by mckeankylej** [comment]
@zimbatm no problem and this is very exciting this is the first package merged into nixos!

**Comment by zimbatm** [comment]
hehe, congrats :)

**Pull request closed without merging**